internal/pkg/agent/application/gateway/fleet/fleet_gateway.go

```diff
@@ -168,12 +182,30 @@ func (f *fleetGateway) worker() {
 				continue
 			}
 
-			actions := make([]fleetapi.Action, len(resp.Actions))
-			for idx, a := range resp.Actions {
-				actions[idx] = a
+			actions := f.queueScheduledActions(resp.Actions)
+			actions, err = f.dispatchCancelActions(actions)
+			if err != nil {
+				f.log.Error(err.Error())
+				// TODO set errMsg? update status?
```

Comment:
Should we do anything differently if a `Cancel` action fails?

Reply:
@michel-laterman I think we would be fine in only logging it, I wouldn't update the status. cc @joshdover